component/http/route.go

```diff
@@ -194,6 +194,61 @@ func (rb *RouteBuilder) Build() (Route, error) {
 	}, nil
 }
 
+// NewFileServer constructor.
+func NewFileServer(path string, assetsDir string, fallbackPath string) *RouteBuilder {
+	var ee []error
+
+	if path == "" {
+		ee = append(ee, errors.New("path is empty"))
+	}
+
+	if assetsDir == "" {
+		ee = append(ee, errors.New("assets path is empty"))
+	} else {
+		_, err := os.Stat(assetsDir)
+		if os.IsNotExist(err) {
+			ee = append(ee, errors.New("assets directory doesn't exist"))
+		} else if err != nil {
+			ee = append(ee, fmt.Errorf("error while checking assets dir: %w", err))
+		}
+	}
+
+	if fallbackPath == "" {
+		ee = append(ee, errors.New("fallback path is empty"))
+	} else {
+		_, err := os.Stat(fallbackPath)
+		if os.IsNotExist(err) {
+			ee = append(ee, errors.New("fallback file doesn't exist"))
+		} else if err != nil {
+			ee = append(ee, fmt.Errorf("error while checking fallback file: %w", err))
+		}
+	}
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		params := httprouter.ParamsFromContext(r.Context())
```

Comment:
Is there a way to avoid httprouter bleeding in here?

Reply:
I am not sure if there's an existing way to do it, we have to reach into it for the `httprouter.ParamsFromContext` which gets the wildcard routing from the underlying router. 

I suppose we can wrap this ourselves in the router abstraction of patron, should I go ahead and fix that?

Reply:
We have an exported `http.ExtractParams` helper to do that which exposes a `map[string]string`. 

It was initially needed for better handling of RawRoutes parameters; could it also help here?

Reply:
Let me check it out and see what I can do!

Reply:
Yep, that did the trick! 💪